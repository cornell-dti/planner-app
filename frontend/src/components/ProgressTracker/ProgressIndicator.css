.ProgressIndicator {
  display: flex;
  align-content: center;
  margin: 1em;
}

.ProgressText {
  text-align: center;
}

.ProgressCountText {
  font-weight: bold;
}

.ProgressOtherText {
  color: #9B9B9B;
}

.ProgressBarContainer {
  flex: auto;
  margin-left: 1em;
  position: relative;
  height: 2em;
  border-radius: 1em;
  background-color: rgba(155, 155, 155, 0.19);
}

.ProgressBarIndicator {
  position: absolute;
  top: 0;
  left: 0;
  height: 2em;
  border-radius: 1em;
  background-color: #7ED321;
}
