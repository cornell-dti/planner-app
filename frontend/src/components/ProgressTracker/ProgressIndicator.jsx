// @flow strict

import React from 'react';
import type { Node } from 'react';
import type { TasksProgress, TasksProgressProps } from '../../util/task-util';
import styles from './ProgressIndicator.css';

type ProgressBarProps = {|
  +progress: TasksProgress;
  +children: Node;
|};

/**
 * A simple progress bar with optional children.
 *
 * @param {TasksProgress} progress the current progress.
 * @param {?Node} children an optional children.
 * @return {Node} the rendered progress bar.
 * @constructor
 */
function ProgressBar({ progress, children }: ProgressBarProps): Node {
  const { completed, all } = progress;
  const percentage = all === 0 ? 0 : completed / all * 100;
  const innerStyle = { width: `${percentage}%` };
  return (
    <div className={styles.ProgressBarContainer}>
      <div className={styles.ProgressBarIndicator} style={innerStyle} />
      {children}
    </div>
  );
}

// eslint-disable-next-line
ProgressBar.defaultProps = { children: null };

/**
 * The progress indicator.
 *
 * @param {WindowSize} windowSize the current window size.
 * @param {TasksProgress} progress the current progress.
 * @return {Node} the rendered progress bar.
 * @constructor
 */
export default function ProgressIndicator({ progress }: TasksProgressProps): Node {
  const { completed, all } = progress;
  return (
    <div className={styles.ProgressIndicator}>
      <div className={styles.ProgressText}>
        <div>{`${completed}/${all}`}</div>
        <div className={styles.ProgressOtherText}>tasks complete</div>
      </div>
      <ProgressBar progress={progress} />
    </div>
  );
}
