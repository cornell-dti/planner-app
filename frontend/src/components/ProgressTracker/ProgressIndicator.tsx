import React, { ReactNode, ReactElement } from 'react';
import { TasksProgressProps } from '../../util/task-util';
import styles from './ProgressIndicator.css';

type ProgressBarProps = TasksProgressProps & { readonly children?: ReactNode };

/**
 * A simple progress bar with optional children.
 */
function ProgressBar(
  { completedTasksCount, allTasksCount, children }: ProgressBarProps,
): ReactElement {
  const percentage = allTasksCount === 0 ? 0 : completedTasksCount / allTasksCount * 100;
  const innerStyle = { width: `${percentage}%` };
  return (
    <div className={styles.ProgressBarContainer}>
      <div className={styles.ProgressBarIndicator} style={innerStyle} />
      {children}
    </div>
  );
}

ProgressBar.defaultProps = { children: null };

/**
 * The progress indicator.
 */
export default function ProgressIndicator(props: TasksProgressProps): ReactElement {
  const { completedTasksCount, allTasksCount } = props;
  const fractionString = `${completedTasksCount}/${allTasksCount}`;
  return (
    <div className={styles.ProgressIndicator}>
      <div className={styles.ProgressText}>
        <div className={styles.ProgressCountText}>{fractionString}</div>
        <div className={styles.ProgressOtherText}>tasks complete</div>
      </div>
      <ProgressBar completedTasksCount={completedTasksCount} allTasksCount={allTasksCount} />
    </div>
  );
}
