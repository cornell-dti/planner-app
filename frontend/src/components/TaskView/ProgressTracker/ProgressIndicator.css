.ProgressIndicator {
  display: flex;
  align-content: center;
  align-items: center;
  justify-content: center;
  margin: 1em;
}

.ProgressText {
  text-align: center;
}

.ProgressCountText {
  font-weight: bold;
}

.ProgressOtherText {
  color: #9B9B9B;
}

.ProgressBarContainer {
  flex: auto;
  position: relative;
  border-radius: 1em;
  background-color: rgba(155, 155, 155, 0.19);
}

.ProgressBarIndicator {
  position: absolute;
  bottom: 0;
  left: 0;
  border-radius: 1em;
  background-color: #7ED321;

  transition: all 0.7s ease-in-out;
}
