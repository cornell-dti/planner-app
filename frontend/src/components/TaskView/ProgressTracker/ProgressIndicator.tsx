import React, { ReactNode, ReactElement, CSSProperties } from 'react';
import { TasksProgressProps } from 'common/lib/util/task-util';
import styles from './ProgressIndicator.module.scss';

type ProgressBarProps = TasksProgressProps & {
  readonly inMobileView: boolean;
  readonly children?: ReactNode;
};

/**
 * A simple progress bar with optional children.
 */
function ProgressBar(
  { completedTasksCount, allTasksCount, inMobileView, children }: ProgressBarProps,
): ReactElement {
  const percentage = allTasksCount === 0 ? 0 : completedTasksCount / allTasksCount * 100;
  const containerStyle: CSSProperties = inMobileView
    ? { height: '2em', marginLeft: '1em' }
    : { width: '2em', marginBottom: '1em' };
  const innerStyle: CSSProperties = inMobileView
    ? { height: '2em', width: `${percentage}%` }
    : { width: '2em', height: `${percentage}%` };
  return (
    <div className={styles.ProgressBarContainer} style={containerStyle}>
      <div className={styles.ProgressBarIndicator} style={innerStyle} />
      {children}
    </div>
  );
}

ProgressBar.defaultProps = { children: null };

type Props = TasksProgressProps & { readonly inMobileView: boolean };

/**
 * The progress indicator.
 */
export default function ProgressIndicator(
  { completedTasksCount, allTasksCount, inMobileView }: Props,
): ReactElement {
  const containerStyle: CSSProperties = inMobileView
    ? { height: '32px' }
    : { flexDirection: 'column-reverse', margin: '0', width: '32px' };
  const textStyle: CSSProperties = inMobileView ? {} : { width: '70px', marginBottom: '8px' };
  const fractionString = `${completedTasksCount}/${allTasksCount}`;
  return (
    <div className={styles.ProgressIndicator} style={containerStyle}>
      <div className={styles.ProgressText} style={textStyle}>
        <div className={styles.ProgressCountText}>{fractionString}</div>
        <div className={styles.ProgressOtherText}>tasks complete</div>
      </div>
      <ProgressBar
        completedTasksCount={completedTasksCount}
        allTasksCount={allTasksCount}
        inMobileView={inMobileView}
      />
    </div>
  );
}
