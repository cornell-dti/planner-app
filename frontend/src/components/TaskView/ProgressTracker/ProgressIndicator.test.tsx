import React from 'react';
import renderer from 'react-test-renderer';
import ProgressIndicator from './ProgressIndicator';

it('Mobile completed ProgressIndicator matches snapshot', () => {
  const tree = renderer
    .create(<ProgressIndicator inMobileView allTasksCount={3} completedTasksCount={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Mobile incompleted ProgressIndicator matches snapshot', () => {
  const tree = renderer
    .create(<ProgressIndicator inMobileView allTasksCount={1} completedTasksCount={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Desktop completed ProgressIndicator matches snapshot', () => {
  const tree = renderer
    .create(<ProgressIndicator inMobileView={false} allTasksCount={3} completedTasksCount={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('Desktop incompleted ProgressIndicator matches snapshot', () => {
  const tree = renderer
    .create(<ProgressIndicator inMobileView={false} allTasksCount={1} completedTasksCount={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
