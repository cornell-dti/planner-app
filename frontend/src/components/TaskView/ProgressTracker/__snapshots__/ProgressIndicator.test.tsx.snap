// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop completed ProgressIndicator matches snapshot 1`] = `
<div
  className="ProgressIndicator"
  style={
    Object {
      "flexDirection": "column-reverse",
      "margin": "0",
      "width": "32px",
    }
  }
>
  <div
    className="ProgressText"
    style={
      Object {
        "marginBottom": "8px",
        "width": "70px",
      }
    }
  >
    <div
      className="ProgressCountText"
    >
      3/3
    </div>
    <div
      className="ProgressOtherText"
    >
      tasks complete
    </div>
  </div>
  <div
    className="ProgressBarContainer"
    style={
      Object {
        "marginBottom": "1em",
        "width": "2em",
      }
    }
  >
    <div
      className="ProgressBarIndicator"
      style={
        Object {
          "height": "100%",
          "width": "2em",
        }
      }
    />
  </div>
</div>
`;

exports[`Desktop incompleted ProgressIndicator matches snapshot 1`] = `
<div
  className="ProgressIndicator"
  style={
    Object {
      "flexDirection": "column-reverse",
      "margin": "0",
      "width": "32px",
    }
  }
>
  <div
    className="ProgressText"
    style={
      Object {
        "marginBottom": "8px",
        "width": "70px",
      }
    }
  >
    <div
      className="ProgressCountText"
    >
      3/1
    </div>
    <div
      className="ProgressOtherText"
    >
      tasks complete
    </div>
  </div>
  <div
    className="ProgressBarContainer"
    style={
      Object {
        "marginBottom": "1em",
        "width": "2em",
      }
    }
  >
    <div
      className="ProgressBarIndicator"
      style={
        Object {
          "height": "300%",
          "width": "2em",
        }
      }
    />
  </div>
</div>
`;

exports[`Mobile completed ProgressIndicator matches snapshot 1`] = `
<div
  className="ProgressIndicator"
  style={
    Object {
      "height": "32px",
    }
  }
>
  <div
    className="ProgressText"
    style={Object {}}
  >
    <div
      className="ProgressCountText"
    >
      3/3
    </div>
    <div
      className="ProgressOtherText"
    >
      tasks complete
    </div>
  </div>
  <div
    className="ProgressBarContainer"
    style={
      Object {
        "height": "2em",
        "marginLeft": "1em",
      }
    }
  >
    <div
      className="ProgressBarIndicator"
      style={
        Object {
          "height": "2em",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`Mobile incompleted ProgressIndicator matches snapshot 1`] = `
<div
  className="ProgressIndicator"
  style={
    Object {
      "height": "32px",
    }
  }
>
  <div
    className="ProgressText"
    style={Object {}}
  >
    <div
      className="ProgressCountText"
    >
      3/1
    </div>
    <div
      className="ProgressOtherText"
    >
      tasks complete
    </div>
  </div>
  <div
    className="ProgressBarContainer"
    style={
      Object {
        "height": "2em",
        "marginLeft": "1em",
      }
    }
  >
    <div
      className="ProgressBarIndicator"
      style={
        Object {
          "height": "2em",
          "width": "300%",
        }
      }
    />
  </div>
</div>
`;
